rust/lance-encoding/src/encoder.rs

```diff
@@ -479,6 +485,28 @@ impl CoreArrayEncodingStrategy {
         }
     }
 
+    fn choose_encoder_for_numeric(
+        arrays: &[ArrayRef],
+        data_type: &DataType,
+        version: LanceFileVersion,
+        field_meta: Option<&HashMap<String, String>>,
+    ) -> Box<dyn ArrayEncoder> {
+        let mut encoder: Option<Box<dyn ArrayEncoder>> = None;
+
+        if version >= LanceFileVersion::V2_1 {
+            if arrays[0].data_type() == data_type {
+                let compressed_bit_width = compute_compressed_bit_width_for_non_neg(arrays);
+                encoder = Some(Box::new(BitpackedForNonNegArrayEncoder::new(
+                    compressed_bit_width as usize,
+                    data_type.clone(),
+                )));
+            } else if let Some(compression) = field_meta.and_then(Self::get_field_compression) {
```

Comment:
When bitpacking is not applicable, check if field itself specifies some compression